### ISSUE TITLE
Project deletion causes division by zero exception.

### ISSUE THREAD

**Issue by umutoztunc** [bounswe/bounswe2018group8#77]
The following signal handler causes division by zero exception whenever a project is deleted. It deletes the bid objects first and just before deleting the actual project object the signal handler kicks in and tries to update the average bid by dividing the total bid amount to the current bid count. Since they are both zero after the bid objects are deleted.

```
@receiver(pre_delete, sender=Bid)
def update_bid_info_on_delete(sender, instance, *args, **kwargs):
    project = instance.project
    total = project.bid_count * project.average_bid
    total -= instance.amount
    project.bid_count -= 1
    project.average_bid = total / project.bid_count
    project.save()
```

In order to resolve this issue, we can either set the average bid to zero or null whenever a project's bid count is decreased to zero.

**Comment by mehmetcalim** [comment]
In my opinion, null is better than zero for average bid whenever a project's bid count is decreaces to zero.

**Comment by umutoztunc** [comment]
I have fixed the bug by setting the average bid to zero instead of null. If it was possible for average bid to be zero and bid count to be nonzero at the same time, then it would be better to use null to distinguish the two cases where bid count is zero while average is zero and bid count greater than zero while average is zero. Since it is impossible for average to be zero without bid count being zero, I decided to simply set the average to zero.
